frontend/lib/api/form/pollingstation/PollingStationFormController.tsx

```diff
@@ -375,89 +296,144 @@ export function PollingStationFormController({
     [currentForm, formState],
   );
 
-  const submitCurrentForm = React.useCallback(
-    (ignoreWarnings?: boolean) => {
-      if (currentForm.current) {
-        const ref: AnyFormReference = currentForm.current;
-
-        //flag this submit to ignore warnings
-        _ignoreWarnings.current = ignoreWarnings ? ref.id : null;
-
-        switch (ref.type) {
-          case "political_group_votes":
-            setValues((old) => ({
-              ...old,
-              political_group_votes: old.political_group_votes.map((pg) => {
-                if (pg.number === ref.number) {
-                  return ref.getValues();
-                }
-                return pg;
-              }),
-            }));
-            break;
-          case "recounted": {
-            const newValues = ref.getValues();
-
-            setValues((old) => {
-              let voters_recounts: VotersRecounts | undefined = undefined;
-
-              if (newValues.recounted) {
-                if (old.recounted === false) {
-                  voters_recounts = {
-                    poll_card_recount: 0,
-                    proxy_certificate_recount: 0,
-                    voter_card_recount: 0,
-                    total_admitted_voters_recount: 0,
-                  };
-                } else {
-                  voters_recounts = old.voters_recounts;
-                }
+  const submitCurrentForm = async (ignoreWarnings?: boolean) => {
+    // React state is fixed within one render, so we update our own copy instead of using setValues directly
+    let newValues: PollingStationValues = values;
+    if (currentForm.current) {
+      const ref: AnyFormReference = currentForm.current;
+
+      //flag this submit to ignore warnings
+      _ignoreWarnings.current = ignoreWarnings ? ref.id : null;
```

Comment:
This flag probably isn't necessary anymore ?

Reply:
Indeed, removed in b8f78c45e214f4ea949d25b7387fbf1b20da2d79.